src/mdoc/model/DeviceResponse.ts

```diff
@@ -81,20 +83,81 @@ export class DeviceResponse {
+    )
+   * It is preferable to use `usingSessionTranscriptBytesForOID4VP` or `usingSessionTranscriptBytesForWebAPI` when possible.
    * @param {Buffer} sessionTranscriptBytes - The sessionTranscriptBytes data to use in the session transcript.
    * @returns {DeviceResponse}
    */
   public usingSessionTranscriptBytes(sessionTranscriptBytes: Buffer): DeviceResponse {
     if (this.sessionTranscriptBytes) {
-      throw new Error('A session transcript has already been set, either with .usingHandover or .usingSessionTranscriptBytes');
+      throw new Error(
+        'A session transcript has already been set, with one of the .usingSessionTranscriptBytesXXX methods',
+      );
     }
     this.sessionTranscriptBytes = sessionTranscriptBytes;
     return this;
   }
 
+  /**
+   * Set the session transcript data to use for the device response as defined in ISO/IEC 18013-7 in Annex B (OID4VP).
+   * This should match the session transcript as it will be calculated by the verifier.
+   * @param {string} mdocGeneratedNonce -
+   * @param {string} clientId -
+   * @param {string} responseUri -
+   * @param {string} verifierGeneratedNonce -
+   * @returns {DeviceResponse}
+   */
+  public usingSessionTranscriptBytesForOID4VP(
+    mdocGeneratedNonce: string,
+    clientId: string,
+    responseUri: string,
+    verifierGeneratedNonce: string,
+  ): DeviceResponse {
+    this.usingSessionTranscriptBytes(
+      cborEncode(
+        DataItem.fromData([
+          null, // deviceEngagementBytes
+          null, // eReaderKeyBytes
+          [mdocGeneratedNonce, clientId, responseUri, verifierGeneratedNonce],
+        ]),
+      ),
+    );
+    return this;
+  }
+
+  /**
+   * Set the session transcript data to use for the device response as defined in ISO/IEC 18013-7 in Annex A (Web API).
+   * This should match the session transcript as it will be calculated by the verifier.
+   * @param {Buffer} deviceEngagementBytes - The device engagement, encoded as a Tagged 24 cbor
+   * @param {Buffer} readerEngagementBytes - The reader engagement, encoded as a Tagged 24 cbor
+   * @param {Buffer} eReaderKeyBytes - The reader ephemeral public key as a COSE Key, encoded as a Tagged 24 cbor
+   * @returns {DeviceResponse}
+   */
+  public usingSessionTranscriptBytesForWebAPI(
+    deviceEngagementBytes: Buffer,
+    readerEngagementBytes: Buffer,
+    eReaderKeyBytes: Buffer,
+  ): DeviceResponse {
+    if (this.sessionTranscriptBytes) {
+      throw new Error(
+        'A session transcript has already been set, either with .usingSessionTranscriptBytesForOID4VP or .usingSessionTranscriptBytesForWebAPI',
+      );
+    }
+    this.usingSessionTranscriptBytes(
+      cborEncode(
+        DataItem.fromData([
+          new DataItem({ buffer: deviceEngagementBytes }),
+          new DataItem({ buffer: eReaderKeyBytes }),
+          Buffer.from(createHash('sha256').update(readerEngagementBytes).digest('hex'), 'hex'),
+        ]),
+      ),
```

Comment:
please make sure that this is defined by the 2023 draft

Reply:
Good callout, in the 2023 draft, `engagementToApp` is simply the `ReaderEngagementBytes`. Fixing this.

Reply:
Fixed in 2ad1bcc